peer/peer.go

```diff
@@ -849,6 +852,35 @@ func (p *Peer) PushAddrMsg(addresses []*wire.NetAddress) ([]*wire.NetAddress, er
 	return msg.AddrList, nil
 }
 
+// PushAddrV2Msg is similar to PushAddrMsg, but for NetAddressV2 addresses.
+func (p *Peer) PushAddrV2Msg(addresses []*wire.NetAddressV2) ([]*wire.NetAddressV2, error) {
+	addressCount := len(addresses)
+
+	// Nothing to send.
+	if addressCount == 0 {
+		return nil, nil
+	}
+
+	msg := wire.NewMsgAddrV2()
+	msg.AddrList = make([]*wire.NetAddressV2, addressCount)
+	copy(msg.AddrList, addresses)
+
+	// Randomize the addresses sent if there are more than the maximum allowed.
+	if addressCount > wire.MaxAddrPerMsg {
```

Comment:
Security - Low: Let's always randomise address order, so that peers can't discover the order of our internal address list.

This could be a security issue if we use our internal address list order to pick our seed peers.